conans/test/functional/generators/visual_studio_multi_test.py

```diff
@@ -0,0 +1,158 @@
+# coding=utf-8
+
+import os
+import platform
+import unittest
+
+from nose.plugins.attrib import attr
+from parameterized import parameterized
+
+from conans import MSBuild, tools
+from conans.client.runner import ConanRunner
+from conans.test.utils.conanfile import MockConanfile, MockSettings
+from conans.test.utils.tools import TestClient
+from conans.test.utils.visual_project_files import get_vs_project_files
+
+main_cpp = r"""#include <hello.h>
+
+int main()
+{
+    hello();
+}
+"""
+
+conanfile_txt = r"""[requires]
+Hello1/0.1@lasote/testing
+[generators]
+{generator}
+"""
+
+hello_conanfile_py = r"""from conans import ConanFile, CMake
+import platform, os, shutil
+
+class HelloConan(ConanFile):
+    name = "Hello1"
+    version = "0.1"
+    settings = "os", "compiler", "arch", "build_type"
+    generators = "cmake"
+    exports = '*'
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.libs = ["hello"]
+        self.cpp_info.debug.defines = ["CONAN_DEBUG"]
+        self.cpp_info.release.defines = ["CONAN_RELEASE"]
+"""
+
+hello_cmake_lists_txt = r"""cmake_minimum_required(VERSION 2.8)
+project(hello)
+
+add_library(hello hello.cpp hello.h)
+set_target_properties(hello PROPERTIES PUBLIC_HEADER "hello.h")
+install(TARGETS hello
+  LIBRARY DESTINATION "lib"
+  ARCHIVE DESTINATION "lib"
+  RUNTIME DESTINATION "lib"
+  PUBLIC_HEADER DESTINATION "include")
+"""
+
+hello_h = r"""#pragma once
+
+void hello_debug();
+void hello_release();
+
+#ifdef CONAN_DEBUG
+#define hello hello_debug
+#endif
+
+#ifdef CONAN_RELEASE
+#define hello hello_release
+#endif
+"""
+
+hello_cpp = r"""#include "hello.h"
+
+#include <iostream>
+
+static void print_arch()
+{{
+#ifdef _M_IX86
+    std::cout << "x86" << std::endl;
+#endif
+#ifdef _M_X64
+    std::cout << "x64" << std::endl;
+#endif
+}}
+
+void hello_release()
+{{
+    std::cout << "Hello Release" << std::endl;
+    print_arch();
+}}
+
+void hello_debug()
+{{
+    std::cout << "Hello Debug" << std::endl;
+    print_arch();
+}}
+"""
+
+
+class VisualStudioMultiTest(unittest.TestCase):
+    @parameterized.expand([("visual_studio", "conanbuildinfo.props"),
+                           ("visual_studio_multi", "conanbuildinfo_multi.props")])
+    @attr('slow')
+    @unittest.skipUnless(platform.system() == "Windows", "Requires MSBuild")
+    def build_vs_project_test(self, generator, props):
+        client = TestClient()
+        files = get_vs_project_files()
+        files["MyProject/main.cpp"] = main_cpp
+        files["conanfile.txt"] = conanfile_txt.format(generator=generator)
+        files["hello/conanfile.py"] = hello_conanfile_py
```

Comment:
Please do first the export (or create) of hello, exclusively.

Then later, define your consumer MyProject project.

Reply:
I am not sure I understand this, export is done below

Reply:
https://github.com/conan-io/conan/pull/4300/files#diff-bbfa689a2acdf6ea90ac7ab7b305af2aR128:
`client.run("export hello lasote/testing")`